core/src/immutable_deserialized_bundle.rs

```diff
@@ -99,12 +111,53 @@ impl ImmutableDeserializedBundle {
             immutable_packets.push(immutable_packet);
         }
 
+        // Assuming tip transactions are SystemProgram transfers with a specific data format
+        let tip_amount = immutable_packets
+            .iter()
+            .map(|packet| Self::extract_tips_from_packet(packet, tip_accounts))
+            .sum::<u64>();
+
         Ok(Self {
             bundle_id: bundle.bundle_id.clone(),
             packets: immutable_packets,
+            tip_amount,
         })
     }
 
+    fn extract_tips_from_packet(
+        packet: &ImmutableDeserializedPacket,
+        tip_accounts: &HashSet<Pubkey>,
+    ) -> u64 {
+        let tx = packet.transaction();
+        let message = tx.get_message();
+        let account_keys = message.message.static_account_keys();
+
+        message
+            .program_instructions_iter()
+            .filter(|(program_id, _)| *program_id == &solana_sdk::system_program::id())
+            .filter_map(|(_, instruction)| {
+                // System program transfer instruction layout:
+                // [0..4]: instruction discriminator (2 = Transfer)
+                // [4..12]: lamports amount (u64)
+                let destination_pubkey = account_keys.get(*instruction.accounts.get(1)? as usize)?;
+                if instruction.data.len() >= 12
+                    && instruction.data[0..4] == [2, 0, 0, 0]
+                    && tip_accounts.contains(destination_pubkey)
+                {
+                    let lamports_bytes = &instruction.data[4..12];
+                    let lamports = u64::from_le_bytes(
+                        lamports_bytes
+                            .try_into()
+                            .expect("Slice with incorrect length"),
+                    );
```

Comment:
If you add `arrayref = { workspace = true }` to the core/cargo.toml, you can then do:

```rust
let lamports = u64::from_le_bytes(*array_ref![&instruction.data, 4, 8]);
```